lib/player.rb

```diff
@@ -1,39 +1,30 @@
-require_relative 'create_ship'
+require_relative 'ships_list'
+require_relative 'ui'
 require_relative 'ship'
+require_relative 'validated_ui'
 
 class Player
 
-  def initialize(grid)
-    @grid = grid
-    @grid_display = GridDisplay.new
-    @create_ship = CreateShip.new
-  end
-
-  def place_ship(x, y, ship, orientation)
-    @grid.mark_ship_positions(x, y, ship, orientation)
-  end
+  attr_reader :name
 
-  def attack(cell_position)
-    ship = @grid.ship_on(cell_position)
-    ship_exist?(ship) ? hit(cell_position, ship) : :water
-  end
-
-  private
-
-  def ship_exist?(ship)
-    ship != nil
+  def initialize(name, grid, validated_ui, ui)
+    @name = name
+    @grid = grid
+    @validated_ui = validated_ui
+    @ui = ui
   end
 
-  def hit(cell_position, ship)
-    ship.register_cells_hit(cell_position)
-    update_sunk_ships_list(ship)
-    :hit
+  def placement_move(player_name, ships_list, ui)
```

Comment:
Hm, I knew it might be confusing... it should describe player's move of placing ship. The method doesn't really place the ships as it only returns a ship which will then be placed on grid by grid itself. Not sure how to call it, honestly